dandischema/models.py

```diff
@@ -58,8 +64,12 @@
 )
 ASSET_UUID_PATTERN = r"^dandiasset:" + UUID_PATTERN
 VERSION_PATTERN = r"\d{6}/\d+\.\d+\.\d+"
-DANDI_DOI_PATTERN = rf"^10.(48324|80507)/dandi\.{VERSION_PATTERN}"
-DANDI_PUBID_PATTERN = rf"^DANDI:{VERSION_PATTERN}"
+# Vendored
+# DANDI_DOI_PATTERN = rf"^10\.(48324|80507)/dandi\.{VERSION_PATTERN}"
+# Unvendored, only with 10. prefix, as likely all would have for datacite
+DANDI_DOI_PATTERN = rf"^10\.\d+/[a-z]+\.{VERSION_PATTERN}"
```

Comment:
@yarikoptic I think this PR is complete except you may want to go further with this line by leveraging the configuration in `conf.py` like what is done for `DANDI_PUBID_PATTERN`. There are two parts in this pattern you may want get through the config (I don't know how appropriately to name them). The `\d+` (or `(48324|80507)` in the vendored version) and the `[a-z]+` (or `dandi` in the vendor version).

Reply:
so we are limited to datacite ATM. I guess (didn't check), all datacite DOIs will start with `10.` (TODO: check)

so I feel like we need some `datacite_doi_id` which e.g. would be that 48324 for one of the dandi instances, and then compose it together with `f"{id_prefix_pattern.lower()}"`,  so then  here indeed it would become similar to `DANDI_PUBID_PATTERN`:

```suggestion
DANDI_DOI_PATTERN = rf"^10\.{DATACITE_DOI_ID}/{ID_PREFIX_PATTERN.lower()}\.{VERSION_PATTERN}"
```

Reply:
> so we are limited to datacite ATM. I guess (didn't check), all datacite DOIs will start with 10. (TODO: check)

It turns out that the `10.` start is not unique to datacite DOIs. It's the beginning for all DOIs. (See https://support.datacite.org/docs/doi-basics and https://en.wikipedia.org/wiki/Digital_object_identifier).

The number sequence that follows `10.` is called "registrant code" in DOI terminology, so I will use the variable name `DOI_REGISTRANT_CODE` instead of `DATACITE_DOI_ID`.

Reply:
thanks! I think there is no need for `DOI_REGISTRANT_CODE` since as you say all DOIs would have it, and thus all datacite DOIs, and since we support only datacite, no need for anything else . Just add `DATACITE_DOI_ID_PATTERN` or alike, since we will use it as pattern right?

Reply:
requested change is applied.